From ad48509d6fe975d7563d4f1c4ded75275cc32465 Mon Sep 17 00:00:00 2001
From: "Edward A. James" <eajames@us.ibm.com>
Date: Thu, 27 Jul 2017 09:46:05 -0500
Subject: [PATCH linux dev-4.10] drivers/fsi: Remove scan from master
 registration

Scanning during master registration is problematic because the gpio fsi
master will be registered during boot if it is present in device-tree
(it's a platform device). That means that the whole fsi scan occurs
during boot, which can cause problems.

Signed-off-by: Edward A. James <eajames@us.ibm.com>
---
 drivers/fsi/fsi-core.c       | 2 --
 drivers/fsi/fsi-master-hub.c | 4 +++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/fsi/fsi-core.c b/drivers/fsi/fsi-core.c
index 9b83f1a..348ed45 100644
--- a/drivers/fsi/fsi-core.c
+++ b/drivers/fsi/fsi-core.c
@@ -927,8 +927,6 @@ int fsi_master_register(struct fsi_master *master)
 		return rc;
 	}
 
-	fsi_master_scan(master);
-
 	return 0;
 }
 EXPORT_SYMBOL_GPL(fsi_master_register);
diff --git a/drivers/fsi/fsi-master-hub.c b/drivers/fsi/fsi-master-hub.c
index 3223a67..b5bb064 100644
--- a/drivers/fsi/fsi-master-hub.c
+++ b/drivers/fsi/fsi-master-hub.c
@@ -288,8 +288,10 @@ static int hub_master_probe(struct device *dev)
 	hub_master_init(hub);
 
 	rc = fsi_master_register(&hub->master);
-	if (!rc)
+	if (!rc) {
+		fsi_master_rescan(&hub->master);
 		return 0;
+	}
 
 	kfree(hub);
 err_release:
-- 
1.8.3.1

